gen/field.go

```diff
@@ -100,12 +103,36 @@ func (f fieldGroupGenerator) DefineStruct(g Generator) error {
 			// optional collections because omitempty doesn't differentiate
 			// between nil and empty collections.
 
+			jsonTags := ""
+			jsonFieldName := f.Name
+
 			if (isStructType(f.Type) || isPrimitiveType(f.Type)) && !f.Required {
-				return fmt.Sprintf("`json:\"%s,omitempty\"`", f.Name)
+				jsonTags += ",omitempty"
+			}
+
+			if f.Required {
+				jsonTags += ",required"
+			}
+
+			goTag := f.Annotations["go.tag"]
+			if goTag != "" {
```

Comment:
Since `goTag` is only used withing the conditional's cope, you can do this:

`if goTag := string.Trim(f.Annotations["go.tag"]); goTag != ""`

I've also threw a string.Trim there. I am not sure if the parser stage already cleans that up. its easy enough to add a test for it though.